src/Reflectify/Reflectify.cs

```diff
@@ -613,18 +611,20 @@ public Reflector(Type typeToReflect, MemberKind kind)
 
     private void LoadProperties(Type typeToReflect, MemberKind kind)
     {
+        var collectedPropertyNames = new HashSet<string>();
```

Comment:
🤔 I guess you meant that this will be GCed much earlier since the instances of `Reflector` are cached?

Reply:
Since the hashset were fields on statically cached instances of `Reflector` they would never be freed as the GC would regard them as reachable.